src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_Warnings.cs

```diff
@@ -388,7 +399,7 @@ private void CheckVacuousComparisons(BoundBinaryOperator tree, ConstantValue con
             }
         }
 
-        private void CheckOr(BoundBinaryOperator node)
```

Comment:
IMHO the name of this method is a bit misleading.  Can we rename it to `CheckIfOr` or add a comment specifying that it only performs checks in the case that it is an `Or` style operator?